### DISCUSSION TITLE
Yound Padawan looking for master jedi to learn SQLalchemy best practices

### DISCUSSION THREAD

**Discussion by fbossiere** [sqlalchemy/sqlalchemy#7023] [Usage Questions]
Hello,

I am working on various projects involving SQLAlchemy and I am starting to really enjoy using this ORM. My problem is that I find it very complex and I am already sure that I don't follow any best practice.

That's why I am looking for someone who really knows well this technology and who would agree to teach me. Of course I am willing to pay for this service, and I can discuss this in private.

Thank you in advance,
Best regards,
Paco

**Comment by CaselIT** [comment]
Hi,

Just a couple of suggestion. I would suggest reading through the [new 1.4/2.0 tutorial](https://docs.sqlalchemy.org/en/14/tutorial/index.html) and also the migration notes regarding [1.4](https://docs.sqlalchemy.org/en/14/changelog/migration_14.html) and [2.0](https://docs.sqlalchemy.org/en/14/changelog/migration_20.html). 
Sadly the documentation is very vast and an not all of it is always up to date

Mike also did some presentations on sqlalchemy 1.4/2 you can find of youtube https://www.youtube.com/watch?v=1Va493SMTcY

Not the master jedi you are looking for, but hopefully it can still be of some help

**Reply by fbossiere** [reply]
Thanks a lot, it's already some very interesting material !

**Reply by zzzeek** [reply]
id also recommend my "SQLAlchemy Session" video at https://www.sqlalchemy.org/library.html#thesqlalchemysessionindepth which tries to walk through the philosophy of the Session object and why it's doing what it does.    most "best practices" kinds of issues seem to revolve around transactions and connections so this gets at how we've chosen to go with that (among many options).

**Reply by CaselIT** [reply]
it seems that from that library page is missing the video you did last year with the initial part of 1.4 migration. I guess it's less relevant now, since the newer one is more up-to-date

**Reply by zzzeek** [reply]
some of the pycon accounts have made the videos not embeddable, which is unsightly, but if you click on them it will take you to youtube where the video should be available.  i basically have to update those links to not use embed for now.  which is the video that's "missing", you mean the pywebconf one i did last year?   i dont need to link that one it wasnt very good :)

**Reply by CaselIT** [reply]
> you mean the pywebconf one i did last year? i dont need to link that one it wasnt very good :)

yes, the one that used `future` before we basically deprecated that :)

**Reply by CaselIT** [reply]
This one https://www.youtube.com/watch?v=sO7FFPNvX2s
